src/Publisher.js

```diff
@@ -0,0 +1,69 @@
+const QueueMessage = require('./QueueMessage')
+
+class Publisher {
+  /**
+   * @param {QueueConnection} queueConnection
+   * @param {Console} logger
+   * @param {String} name
+   */
+  constructor (queueConnection, logger, name) {
+    this._connection = queueConnection
+    this._logger = logger
+    this.name = name
+  }
+
+  /**
+   * @param {String} message
+   * @param {String} correlationId
+   * @return {Promise}
+   */
+  send (message, correlationId) {
+    let options = {}
+    let channel
+    let param
+
+    if (correlationId) {
+      options.correlationId = correlationId
+    }
+
+    try {
+      param = JSON.stringify(new QueueMessage('ok', message))
+    } catch (err) {
+      this._logger.error('CANNOT PUBLISH MESSAGE %s #%s', this.name, err)
+      throw err
+    }
+
+    return this._connection.getChannel().then((ch) => {
+      channel = ch
+      return channel.assertExchange(this.name, 'fanout', {durable: true})
+    }).then(() => {
+      let isResolved = false
+
+      return new Promise((resolve, reject) => {
+        let callResolve = () => {
```

Comment:
ez a többibe sem kell
itt is tegyük át a stringify-t a promise-ba
nézzük meg, hogy származtatással mivan